### PR TITLE
Remove @wordpress/data dependency in the email input iframe file

### PR DESCRIPTION
**Pull request by hsingyuc**
#### Changes proposed in this Pull Request
@frosso found there are 15 more JS files (~500kB more) added on the shortcode checkout page due to introducing `@wordpress/data` [in the email input iframe file](https://github.com/Automattic/woocommerce-payments/blob/85d18e76cf531f77e47ecf4bbe43b89cc2aae5d1/client/checkout/woopay/email-input-iframe.js#L14).   This PR removes the dependency and check if there's WooPayments on the Blocks or Classic checkout page instead.

#### Testing instructions
**Test with WooCommerce Bookings**
1. Install and activate the WC Bookings plugin
2. Check the "Requires confirmation?" box
3. Use an existing WooPay shopper email on the checkout page
4. The login session iframe should not be opened

**Test Blocks checkout page**
1. As an existing WooPay shopper, add a product to the cart
2. Go to Blocks checkout page and the login session iframe should be opened
3. Go to Classic checkout page and the login session iframe should be opened

-------------------

- [X] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8561` or branch name `remove/dependency-wordpress/data` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=remove/dependency-wordpress/data) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 75e008cc4376ed2585ea3f2eac0cf412461e2b81
- Build time: 2024-04-10 03:40:17 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/checkout/woopay/email-input-iframe.js [hunk 1]
```diff
@@ -617,13 +616,14 @@ export const handleWooPayEmailInput = async (
 	} );
 
 	if ( ! customerClickedBackButton ) {
-		const paymentMethods = await select(
-			'wc/store/payment'
-		).getAvailablePaymentMethods();
-
-		const hasWCPayPaymentMethod = paymentMethods.hasOwnProperty(
-			'woocommerce_payments'
+		const getWCPayElementOnBlocks = document.getElementById(
```
Comment (frosso) [inline client/checkout/woopay/email-input-iframe.js]: The `get*` prefix could infer that `getWCPayElementOnBlocks` (and `getWCPayElementOnClassic`) are functions, when in reality they're constants.
Maybe `hasWcPayElementOnBlocks` and `hasWcPayElementOnShortcode` could be some alternatives?

Let me know your thoughts!

**Review by frosso — APPROVED** [review]
The changes read well, thanks for fixing the dependency! I have a small suggestion on naming.

File: client/checkout/woopay/email-input-iframe.js (hunk 1 above)
Comment (hsingyuc) [reply client/checkout/woopay/email-input-iframe.js]: Thank you for reviewing, @frosso!  You are right.  Updated here [a5b4aa0](https://github.com/Automattic/woocommerce-payments/pull/8561/commits/a5b4aa04140fb72b44d6dc1f893f3af04c5bdea3).

File: client/checkout/woopay/email-input-iframe.js [hunk 2]
```diff
@@ -617,13 +616,14 @@ export const handleWooPayEmailInput = async (
 	} );
 
 	if ( ! customerClickedBackButton ) {
-		const paymentMethods = await select(
-			'wc/store/payment'
-		).getAvailablePaymentMethods();
-
-		const hasWCPayPaymentMethod = paymentMethods.hasOwnProperty(
-			'woocommerce_payments'
+		const hasWcPayElementOnBlocks = document.getElementById(
```
Comment (asumaran) [inline client/checkout/woopay/email-input-iframe.js:619]: ~This seems to have broken WooPay automatic redirection on short code checkout page. https://github.com/Automattic/woocommerce-payments/issues/8398. The selector doesn't exist in short code checkout.~

Nope. I just noticed the `hasWcPayElementOnShortcode` selector.

Comment (asumaran) [reply client/checkout/woopay/email-input-iframe.js:619]: I'm working on a fix 👍

**Pull request merged**